core/trino-main/src/main/java/io/trino/cost/StageRuntimeInfoProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.cost;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.sql.planner.PlanFragment;
+import io.trino.sql.planner.plan.PlanFragmentId;
+import io.trino.sql.planner.plan.PlanNode;
+
+import java.util.Map;
+
+import static io.trino.execution.scheduler.faulttolerant.OutputStatsEstimator.OutputStatsEstimateResult;
+import static java.util.Objects.requireNonNull;
+
+public class StageRuntimeInfoProvider
+{
+    public static final StageRuntimeInfoProvider EMPTY = new StageRuntimeInfoProvider(ImmutableMap.of(), ImmutableMap.of());
+
+    private final Map<PlanFragmentId, OutputStatsEstimateResult> runtimeOutputStats;
+    private final Map<PlanFragmentId, PlanFragment> planFragments;
+
+    public StageRuntimeInfoProvider(
+            Map<PlanFragmentId, OutputStatsEstimateResult> runtimeOutputStats,
+            Map<PlanFragmentId, PlanFragment> planFragments)
+    {
+        this.runtimeOutputStats = requireNonNull(runtimeOutputStats, "runtimeOutputStats is null");
+        this.planFragments = requireNonNull(planFragments, "planFragments is null");
+    }
+
+    public OutputStatsEstimateResult getRuntimeOutputStats(PlanFragmentId planFragmentId)
+    {
+        return runtimeOutputStats.getOrDefault(planFragmentId, OutputStatsEstimateResult.unknown());
+    }
+
+    public PlanNodeStatsEstimate getEstimatedStats(StatsProvider statsProvider, PlanFragmentId planFragmentId)
```

Comment:
Having `getEstimatedStats` in `StageRuntimeInfoProvider` seem unexpected. Why do we need to make a call to `StatsProvider` via `StageRuntimeInfoProvider`. Can we directly use `StageRuntimeInfoProvider` and `StatsProvider` in `RemoteSourceStatsRule` where `getEstimatedStats` is used?